drone/src/agent/mod.rs

```diff
@@ -159,7 +163,7 @@ pub async fn run_agent(agent_opts: AgentOptions) -> NeverResult {
 
     nats.publish(&request).await?;
 
-    let executor = Executor::new(docker, db, nats.clone(), ip, cluster.clone());
+    let executor = Executor::new(docker, db.clone(), nats.clone(), ip, cluster.clone());
```

Comment:
Is it possible (or even preferable?) to pass `db` by reference here instead of cloning?

Reply:
The clone here is cheap, because the underlying data is an `Arc<...>`. We could probably technically get away with a reference here because we don't need a mutable version of the data and only need it in this function, but I'm not sure.